File: services/common/src/main/java/com/visor/school/common/api/ApiResponse.java
package com.visor.school.common.api;

import java.time.Instant;

/**
 * Standard API response wrapper
 */
public record ApiResponse<T>(
        boolean success,
        T data,
        String message,
        Instant timestamp
) {
    public ApiResponse(boolean success, T data, String message) {
        this(success, data, message, Instant.now());
    }

    public static <T> ApiResponse<T> success(T data) {
        return new ApiResponse<>(true, data, null, Instant.now());
    }

    public static <T> ApiResponse<T> success(T data, String message) {
        return new ApiResponse<>(true, data, message, Instant.now());
    }

    public static <T> ApiResponse<T> error(String message) {
        return new ApiResponse<>(false, null, message, Instant.now());
    }
}

File: services/common/src/main/kotlin/com/visor/school/common/api/ApiResponse.kt
package com.visor.school.common.api

import java.time.Instant

/**
 * Standard API response wrapper
 */
data class ApiResponse<T>(
    val success: Boolean,
    val data: T? = null,
    val message: String? = null,
    val timestamp: Instant = Instant.now()
) {
    companion object {
        @JvmOverloads
        @JvmStatic
        fun <T> success(data: T, message: String? = null): ApiResponse<T> {
            return ApiResponse(true, data, message)
        }

        @JvmStatic
        fun <T> error(message: String): ApiResponse<T> {
            return ApiResponse(false, null, message)
        }
    }
}

